### ISSUE TITLE
PyGMO minimizer  expects alrogrithm name or instance?

### ISSUE THREAD

**Issue by volodymyrss** [giacomov/3ML#211]
PyGMO minimizer  expects alrogrithm name or instance?..

```
/3ml/3ML/threeML/minimizer/minimization.py in contours(self, param_1, param_1_minimum, param_1_maximum, param_1_n_steps, param_2, param_2_minimum, param_2_maximum, param_2_n_steps, progress, **options)
    939                                      "Perform the fit before running contours to remove this warnings.")
    940 
--> 941             pr = ProfileLikelihood(self, fixed_parameters)
    942 
    943             if n_dimensions == 1:

/3ml/3ML/threeML/minimizer/minimization.py in __init__(self, minimizer_instance, fixed_parameters)
    279             if self._original_minimizer.algorithm_name is not None:
    280 
--> 281                 self._optimizer.set_algorithm(self._original_minimizer.algorithm_name)
    282 
    283         else:

/3ml/3ML/threeML/minimizer/pagmo_minimizer.py in set_algorithm(self, algorithm_instance)
     93     def set_algorithm(self, algorithm_instance):
     94 
---> 95         assert isinstance(algorithm_instance, PyGMO.algorithm._algorithm._base), "The algorithm must be an " \
     96                                                                                  "instance of a PyGMO algorithm"
     97 

AssertionError: The algorithm must be an instance of a PyGMO algorithm
```

**Comment by grburgess** [comment]
You should make an instance:

```python

import PyGMO.algorithm as algo

a=algo.de(gen=50)


nl = algo.nlopt_cobyla()
```

**Comment by volodymyrss** [comment]
The call to `set_algorithm` is from inside 3ml, in ProfileLikelihood, see the exception above.
It seems dangerous to change it there. It should work with all minimizers.
So instead I, for myself, made the pagmo minimizer accept both algorithms and their names as strings.

I am not sure that this is best.

Most of all, I do not want to make a pull request before it is possible to make travis unit tests for pagmo. It is clearly not completed and needs integration testing.

Actually, you can not even see this exception unless you fix #209.

**Comment by giacomov** [comment]
(please reopen if still relevant)
